code/game/objects/items/devices/flashlight.dm

```diff
@@ -366,8 +366,8 @@
 		activate()
 
 /obj/item/device/flashlight/glowstick/proc/activate()
-	on = TRUE
-	START_PROCESSING(SSobj, src)
+	if(on)

```

Comment:
this wasn't quite what I said.
what you had before was correct, except it all needed wrapping under an ```if(!on)```, that's all, so:
```
/obj/item/device/flashlight/glowstick/proc/toggle_active()
    if(!on)
        on = TRUE
        START_PROCESSING(SSobj, src)
```
If you wanted to, you could even add an else that makes it toggle off, and rename the proc to ```toggle_active()```, and then add a force argument to it, so:

```
/obj/item/device/flashlight/glowstick/proc/toggle_active(force)
    if(!isnull(force))
        on = force
    else
        on = !on
    if(on)
        START_PROCESSING(SSobj, src)
    else
        STOP_PROCESSING(SSobj, src)
```

then wherever you want it DEFINITELY on, you do ```toggle_active(TRUE)``` and definitely off ```toggle_active(FALSE)``` and if you just want it to... well, "toggle", ```toggle_active()```